### ISSUE TITLE
Get rid of react warning : componentWillMount has been renamed, and is not recommended for use.

### ISSUE THREAD

**Issue by hannes-o-brunner** [salesforce/design-system-react#2386]
@salesforce/design-system-react@0.10.12

React 16.9 emits the following warning:

```
Warning: componentWillMount is deprecated and will be removed in the next major version. Use componentDidMount instead. As a temporary workaround, you can rename to UNSAFE_componentWillMount.

Learn more about this warning here:
https://fb.me/react-async-component-lifecycle-hooks
```

**Comment by tathagat2006** [comment]
@hannes-o-brunner Are you working on this issue? Else I would like to work on this one.

**Comment by hannes-o-brunner** [comment]
No, not working on it, you can take it

**Comment by interactivellama** [comment]
Sounds awesome! Thank you. This PR may be helpful https://github.com/salesforce/design-system-react/pull/2245

**Comment by tathagat2006** [comment]
Great! Thanks @hannes-o-brunner @interactivellama

**Comment by interactivellama** [comment]
I believe this is fixed by #2830
